dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/statelisteners/ElasticListener.java

```diff
@@ -262,6 +269,32 @@ public void afterBulk(long executionId, BulkRequest request,
         }
     }
 
+    /**
+     * Configures the builder for the ES bulk processor. The default settings were causing AWS
+     * ES 429 (too many requests) errors with our current prod data. Drop the default size
+     * and increase the time between retries. Environment variables are there for emergency
+     * overrides, but current settings work when tested.
+     *
+     * See https://ucsc-cgl.atlassian.net/browse/SEAB-3829
+     * @param builder
+     */
+    private void configureBulkProcessorBuilder(Builder builder) {
+        // Default is 5MB
+        final int bulkSizeKb = getEnv("ESCLIENT_BULK_SIZE_KB", 2500);
+        builder.setBulkSize(new ByteSizeValue(bulkSizeKb, ByteSizeUnit.KB));
+
+        // Defaults are 50ms, 8 retries (leaving number of retries the same).
+        final int initialDelayMs = getEnv("ESCLIENT_BACKOFF_INITIAL_DELAY", 500);
+        final int maxNumberOfRetries = getEnv("ESCLIENT_BACKOFF_RETRIES", 8);
+        builder.setBackoffPolicy(BackoffPolicy.exponentialBackoff(TimeValue.timeValueMillis(
+            initialDelayMs), maxNumberOfRetries));
+    }
+
+    private int getEnv(final String name, final int defaultValue) {
+        final String envValue = System.getenv(name);
+        return envValue == null ? defaultValue : Integer.parseInt(envValue);
```

Comment:
If the env var is set, but not to a parsable integer value, this line will throw a `NumberFormatException` and will abort the bulk indexing process (I think) .  Maybe use the default value value in that case?

Reply:
9d51c7e4059bdf2b18d79052b1f140117e35be8e